mcstatus/server.py

```diff
@@ -28,8 +23,13 @@ def ensure_valid_ip(host: object, port: object):
         raise TypeError(f"Port must be an integer port number, got {type(port)} ({port})")
     if port > 65535 or port < 0:
         raise ValueError(f"Port must be within the allowed range (0-2^16), got {port}")
-    if not VALID_HOSTNAME_REGEX.fullmatch(host):
-        raise ValueError(f"Invalid host address, {host!r} (doesn't match the required pattern)")
+    try:
+        # Check if `host` is valid, this is a more accurate way than regex or like this.
+        # `gethostbyname` makes request to dns, and get number ip, so this have 100% accuracy
+        # BUT it is triggers to not registered domains, which can be ignored
+        gethostbyname(host)
```

Comment:
SRV records can be used for Minecraft servers without a corresponding A record, which would break this.